### PR TITLE
Feature/sp16 implement index tree rss dev23

### PR DESCRIPTION
**Pull request by weko3-dev21**
本ストーリーは、Index Listに表示されるRSSアイコンを押下し、インデックス単位でWEKO RSS配信新着情報を取得可能にする。

### CODE REVIEW

File: modules/weko-index-tree/weko_index_tree/utils.py [hunk 1]
```diff
@@ -328,3 +331,52 @@ def get_admin_coverpage_setting():
...
+
+    Arguments:
+        start_date {string} -- start date
+        end_date {string} -- end date
+
+    Returns:
+        dictionary -- elastic search data
+
+    """
+    current_app.logger.debug(index_ids)
```
Comment (weko3-dev21) [inline modules/weko-index-tree/weko_index_tree/utils.py]: Does   the necessary information be output to the log     If you catch an error, you need to keep a log that can identify the cause   of the error.

delete it

File: modules/weko-index-tree/weko_index_tree/utils.py [hunk 2]
```diff
@@ -328,3 +331,52 @@ def get_admin_coverpage_setting():
...
+    path = dict()
+    result = []
+    for index in index_ids:
+        if index.pid == 0:
+            path[str(index.cid)] = str(index.cid)
+            result.append(path[str(index.cid)])
+        else:
+            parent_path = path.get(str(index.pid)) or ""
+            path[str(index.cid)] = (parent_path + "/" + str(index.cid)) if parent_path != "" else "" + str(index.cid)
+            result.append(path[str(index.cid)])
```
Comment (weko3-dev21) [inline modules/weko-index-tree/weko_index_tree/utils.py]: replace by 
parent_path = path.get(str(index.pid)) or ""
            path[str(index.cid)] = (parent_path + "/" + str(index.cid)) if parent_path != "" else "" + str(index.cid)
            result.append(path[str(index.cid)])

File: modules/weko-gridlayout/weko_gridlayout/utils.py [hunk 3]
```diff
@@ -644,8 +645,14 @@ def get_ES_result_by_date(start_date, end_date):
     records_search = RecordsSearch()
     records_search = records_search.with_preference_param().params(
         version=False)
-    search_instance, qs_kwargs = item_search_factory(
-        None, records_search, start_date, end_date)
-    search_result = search_instance.execute()
-    rd = search_result.to_dict()
-    return rd
+    records_search._index[0] = current_app.config['SEARCH_UI_SEARCH_INDEX']
```
Comment (weko3-dev21) [inline modules/weko-gridlayout/weko_gridlayout/utils.py]: Do   you use a magic number?
please define a new config = 0 and use it !

File: modules/weko-index-tree/weko_index_tree/utils.py [hunk 4]
```diff
@@ -328,3 +331,52 @@ def get_admin_coverpage_setting():
...
+        start_date {string} -- start date
+        end_date {string} -- end date
+
+    Returns:
+        dictionary -- elastic search data
+
+    """
+    records_search = RecordsSearch()
+    records_search = records_search.with_preference_param().params(version=False)
+    records_search._index[0] = current_app.config['SEARCH_UI_SEARCH_INDEX']
```
Comment (weko3-dev21) [inline modules/weko-index-tree/weko_index_tree/utils.py]: Do you use a magic number?
please define a new config = 0 and use it !

File: modules/weko-index-tree/weko_index_tree/utils.py [hunk 5]
```diff
@@ -328,3 +331,52 @@ def get_admin_coverpage_setting():
...
+
+def generate_path(index_ids):
+    """Get data from elastic search.
+
+    Arguments:
+        start_date {string} -- start date
+        end_date {string} -- end date
+
+    Returns:
+        dictionary -- elastic search data
```
Comment (weko3-dev21) [inline modules/weko-index-tree/weko_index_tree/utils.py]: wrong comment

File: modules/weko-index-tree/weko_index_tree/utils.py (hunk 1 above)
Comment (weko3-dev23) [reply modules/weko-index-tree/weko_index_tree/utils.py]: Thank, It had been fixed.

File: modules/weko-gridlayout/weko_gridlayout/utils.py (hunk 3 above)
Comment (weko3-dev23) [reply modules/weko-gridlayout/weko_gridlayout/utils.py]: Thank, It had been fixed.

File: modules/weko-index-tree/weko_index_tree/utils.py (hunk 5 above)
Comment (weko3-dev23) [reply modules/weko-index-tree/weko_index_tree/utils.py]: Thank, It had been fixed.

File: modules/weko-search-ui/weko_search_ui/query.py [hunk 6]
```diff
@@ -750,3 +750,67 @@ def _get_query(start_date, end_date):
...
     return search, urlkwargs
+
+    
+def item_search_by_list_index_id(self, search, list_index_id):
+    """Factory for opensearch.
+
+    :param self:
+    :param search:
+    :return:
+    """
```
[the remark this answers is not in the document]
Comment (weko3-dev23) [reply modules/weko-search-ui/weko_search_ui/query.py]: Thank, It had been fixed.

**Pull request closed without merging**